src/scry/log.cr

```diff
@@ -1,20 +1,37 @@
 require "logger"
-require "tempfile"
+require "./client"
+require "./protocol/log_message_params"
 
 module Scry
   class Log
     private def self.initialize_logger
-      tmpfile = Tempfile.new("scry.out")
-      log_file = File.open(tmpfile.path, "w")
-      logger = Logger.new(log_file)
-      logger.level = Logger::INFO
-      logger
+      logger = ClientLogger.new(Client.instance.io)
     end
 
     @@logger : Logger = initialize_logger
 
     def self.logger
       @@logger
     end
```

Comment:
Note: `class_getter logger` does the same thing ;)

(or even `class_property` now that you also have a setter below)